File: 프로그래머스/0/181847. 0 떼기/0 떼기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char* solution(const char* n_str) {
    int i = 0;
    while (n_str[i] == '0')
    {
        i++;
    }
    
    // n_str을 복사하는데 +i 위치에서 복사
    char* result = strdup(n_str + i);
    return result;
}

File: 프로그래머스/0/181847. 0 떼기/0 떼기.java
class Solution {
    public String solution(String n_str) {
        //앞부분에 해당하는 문자열을 뒷부분의 문자열로 교체하는 메서드
        String answer = n_str.replaceFirst("^0+", "");
        // ^ : 시작부분부터 | 0+ : 연속된 하나 이상의 0을 > "" : 빈 문자열로 교체 (삭제)
        return answer;
    }
}